### PR TITLE
[#309] QF to callables at conditionals on for()

### PR DESCRIPTION
**Pull request by rentalhost**
Following #309 

* [x] Improved the code-style (eg. alignment, _space for the code to breathe_, etc.);
* [x] Dropped the `messageExternal` (because it is from an old code, [probably](https://github.com/kalessil/phpinspectionsea/issues/309#issuecomment-302102714));
* [x] Added UTs with full coverage;
* [x] Implemented the QF, QFTs and full coverage too;
* [x] Replaced the message to a more generic one, once that the QF will do the work from now;
* [x] Unknow variables are renamed to `$loopsMax` instead of `$loopsCount` (to be compatible with the "existent definition", as in `$iMax`);
* [ ] Text review is welcome (_but I not changed too much, I guess_);

### CODE REVIEW

**Comment by kalessil** [comment]
This is next one for a merge, the merge is coming this week.

File: src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/CallableInLoopTerminationConditionInspector.java [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+ * (c) Vladimir Reznichenko <kalessil@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
 public class CallableInLoopTerminationConditionInspector extends BasePhpInspection {
-    private static final String messagePattern  = "'for (%existingInit%%newInit%; %newCheck%; ...)' should be used for better performance.";
-    private static final String messageExternal = "Callable result should be stored outside of the loop for better performance.";
+    private static final String message = "Avoid callables in loop conditionals for better performance.";
```
Comment (kalessil) [inline src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/CallableInLoopTerminationConditionInspector.java]: @rentalhost: that's less specific reporting as it was before, is it easier to understand due to introducing QF?

File: src/test/resources/fixtures/semanticalAnalysis/callable-inloop-condition.fixed.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (kalessil) [inline src/test/resources/fixtures/semanticalAnalysis/callable-inloop-condition.fixed.php]: Awesome, thanks for great UTs!

File: src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/CallableInLoopTerminationConditionInspector.java (hunk 1 above)
Comment (rentalhost) [inline src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/CallableInLoopTerminationConditionInspector.java]: It avoid tooltips with codes. making it more clear to understand, while QF do the job.

Comment (kalessil) [inline src/main/java/com/kalessil/phpStorm/phpInspectionsEA/inspectors/semanticalAnalysis/CallableInLoopTerminationConditionInspector.java]: Ok, I'll give it a test-drive with the generalized message

**Pull request merged**